rsmtool/rsmpredict.py

```diff
@@ -164,24 +205,53 @@ def fast_predict(
     # now compute the raw prediction for the given features
     df_predictions = modeler.predict(df_processed_features)
 
-    # compute scaled prediction if the scaling parameters are available
-    if train_predictions_mean and train_predictions_sd and h1_mean and h1_sd:
+    # compute scaled predictions if requested
+    if scale:
+        if train_predictions_mean is None:
+            train_predictions_mean = modeler.train_predictions_mean
+        if train_predictions_sd is None:
+            train_predictions_sd = modeler.train_predictions_sd
+        if h1_mean is None:
+            h1_mean = modeler.h1_mean
+        if h1_sd is None:
+            h1_sd = modeler.h1_sd
         df_predictions["scale"] = (
             (df_predictions["raw"] - train_predictions_mean) / train_predictions_sd
         ) * h1_sd + h1_mean
+    elif any(
+        arg is not None for arg in [train_predictions_mean, train_predictions_sd, h1_mean, h1_sd]
+    ):
+        raise ValueError(
+            "train_predictions_mean/train_predictions_sd/h1_mean/h1_sd cannot be "
+            "specified when scale=False"
+        ) from None
 
     # drop the spkitemid column since it's not needed from this point onwards
     df_predictions.drop("spkitemid", axis="columns", inplace=True)
 
-    # trim both raw and scale predictions if ``trim_min`` and ``trim_max`` were specified
-    if trim_min and trim_max:
+    # trim both raw and scaled predictions if requested
+    if trim:
+        default_trim_tolerance = 0.4998
+        if trim_tolerance is None:
+            try:
+                trim_tolerance = modeler.trim_tolerance
+            except AttributeError:
+                trim_tolerance = default_trim_tolerance
+        if trim_min is None:
+            trim_min = modeler.trim_min
+        if trim_max is None:
+            trim_max = modeler.trim_max
         for column in df_predictions.columns:
             df_predictions[f"{column}_trim"] = preprocessor.trim(
                 df_predictions[column], trim_min, trim_max, trim_tolerance
             )
             df_predictions[f"{column}_trim_round"] = np.rint(
                 df_predictions[f"{column}_trim"]
             ).astype("int64")
+    elif any(arg is not None for arg in [trim_tolerance, trim_min, trim_max]):
```

Comment:
Not to go into a validation rabbit hole, but perhaps we should consider adding validation for the opposite case as well? If backward compatibility is going to be possible, someone might specify trim=True  expecting one of the older models to provide min/max values when it can't. Instead of letting it give the unsupported operand error in preprocessor.trim, it might be good to give a more informative error message before it?

Reply:
I think it would have resulted in an AttributeError (now a ValueError, though). I believe it should be somewhat easy to understand what's going on either way.